sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/schema/BeamTableUtils.java

```diff
@@ -137,6 +138,12 @@ public static Object autoCastField(Schema.Field field, Object rawObj) {
           throw new UnsupportedOperationException(
               String.format("Column type %s is not supported yet!", type));
       }
+    } else if (type.getTypeName().isPrimitiveType()) {
+      if (TypeName.BYTES.equals(type.getTypeName()) && rawObj instanceof ByteString) {
+        return ((ByteString) rawObj).getBytes();
+      }
+      throw new UnsupportedOperationException(
+          String.format("Column type %s is not supported yet!", type));
```

Comment:
@11moon11 it looks like this broke PubsubJsonIT: https://builds.apache.org/job/beam_PostCommit_SQL/3812

```
java.lang.UnsupportedOperationException: Column type FieldType{typeName=BOOLEAN, nullable=true, logicalType=null, collectionElementType=null, mapKeyType=null, mapValueType=null, rowSchema=null, metadata={}} is not supported yet!
	at org.apache.beam.sdk.extensions.sql.impl.schema.BeamTableUtils.autoCastField(BeamTableUtils.java:146)
	at org.apache.beam.sdk.extensions.sql.impl.rel.BeamValuesRel.lambda$tupleToRow$0(BeamValuesRel.java:95)
	at java.util.stream.IntPipeline$4$1.accept(IntPipeline.java:250)
```

Do you think it would be easy to fix or should we roll back?

Reply:
Seems like an easy fix is definitely possible. Just need to add a BOOLEAN case, or remove this UnsupportedOperationException and let it (and other non-numeric primitive types) fall through.

Reply:
Created a PR #10681, which removes this exception.